app/controllers/login_controller.rb

```diff
@@ -43,6 +50,37 @@ def refresh
     session[:oauth_response] = JSON.parse(response)
 
     # Redirect the user's browser to the intended home page
-    redirect_to users_home_path
+    redirect_to users_home_path, success: "You have successfully refreshed your access token!"
+
+    rescue RestClient::ExceptionWithResponse
+      if session[:oauth_response]
+        redirect_to users_home_path, danger: "Something Went Wrong. Please Refresh your Access Token and Try Again"
```

Comment:
My comments regarding the above error messages also apply to this one and the next one (on line 59).